indexer/feature_meta.hpp

```diff
@@ -97,43 +126,28 @@ namespace feature
       {
         ar.Read(header, sizeof(header));
         ar.Read(buffer, header[1]);
-        m_metadata[ToType(header[0] & 0x7F)].assign(buffer, header[1]);
+        m_metadata[header[0] & 0x7F].assign(buffer, header[1]);
       } while (!(header[0] & 0x80));
     }
 
-    template <class ArchiveT> void Serialize(ArchiveT & ar) const
-    {
-      uint8_t const sz = m_metadata.size();
-      WriteToSink(ar, sz);
-      for (auto const & it : m_metadata)
-      {
-        WriteToSink(ar, static_cast<uint8_t>(it.first));
-        utils::WriteString(ar, it.second);
-      }
-    }
+  private:
+    enum { kMaxStringLength = 255 };
+  };
 
-    template <class ArchiveT> void Deserialize(ArchiveT & ar)
+  class AddressData : public MetadataBase
+  {
+  public:
+    enum EType
     {
-      uint8_t const sz = ReadPrimitiveFromSource<uint8_t>(ar);
-      ASSERT_LESS_OR_EQUAL(sz, FMD_COUNT, ());
-
-      for (size_t i = 0; i < sz; ++i)
-      {
-        EType const key = ToType(ReadPrimitiveFromSource<uint8_t>(ar));
-        string value;
-        utils::ReadString(ar, value);
-        m_metadata.insert(make_pair(key, value));
-      }
-    }
+      FAD_PLACE = 1,
```

Comment:
In accordance with our style guide there're no need in E for EType and enum values must be named as TYPE_PLACE, TYPE_STREET and TYPE_POSTCODE.